### PR TITLE
Propogate user not found error from authenticater.

### PR DESCRIPTION
**Pull request by Joerger**
4th request in #6225

### CODE REVIEW

File: lib/auth/grpcserver.go [hunk 1]
```diff
@@ -2229,6 +2229,8 @@ func (g *GRPCServer) authenticate(ctx context.Context) (*grpcContext, error) {
 		// between connection failed and access denied
 		if trace.IsConnectionProblem(err) {
 			return nil, trace.ConnectionProblem(err, "[10] failed to connect to the database")
+		} else if trace.IsNotFound(err) {
+			return nil, trace.Wrap(err, "[10] access denied")
```
Comment (awly) [inline lib/auth/grpcserver.go:2233]: will this show up as `[10] access denied` on the client side, or will the original error be included?

Comment (Joerger) [reply lib/auth/grpcserver.go:2233]: The original error is included. It looked something like `user "teleport" is not found`.

Comment (awly) [reply lib/auth/grpcserver.go:2233]: Cool, please leave a comment here to mention it.
It's easy to assume that the error is masked (the `trace` package is pretty inconsistent about it).

**Review by awly — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**